### ISSUE TITLE
[Feature Request]: Automatic Periodic Library Backups

### ISSUE THREAD

**Issue by CyanVoxel** [TagStudioDev/TagStudio#439]
### Checklist

- [X] I am using an up-to-date version.
- [X] I have read the [documentation](https://github.com/TagStudioDev/TagStudio/blob/main/doc/index.md).
- [X] I have searched existing [issues](https://github.com/TagStudioDev/TagStudio/issues).

### Description

Currently, backups have to be saved manually which introduces an extra step for the user to remember. If the user forgets to save a backup, then the backup feature can't help them.

### Solution

Periodically (every few minutes, and/or triggered on saving) TagStudio could trigger the "Save Library Backup" feature to keep around automatic backups. Ideally, these could be combined with a "Load Library Backup" feature that sets it as the main `ts_library.json/sqlite` file while keeping the current save file as an additional backup.

There would probably need to be some options for setting how often backups are made if at all, and either a cutoff for how many backups are allowed to be made automatically before older backups get deleted (this would need to be approached very carefully, too).

### Alternatives

Autosaving could potentially help, but this will be made obsolete by #332.

**Comment by Cool-Game-Gev** [comment]
@CyanVoxel do we have any updates on this?

**Comment by CyanVoxel** [comment]
> @CyanVoxel do we have any updates on this?

No, I opened up this feature request but I have not been the one developing the feature. Also my focus has been on higher priority library systems since #332 was merged.

**Comment by Cool-Game-Gev** [comment]
@CyanVoxel Thanks!
